docs/further_useful_features.rst

```diff
@@ -0,0 +1,459 @@
+    Although using other iterables such as lists also works, doing so is counter
+    to the intent of the ``iter`` backend (lists would be appropriate in ``eager``).
+
+
+Writing a dataset with a secondary index:
+
+.. ipython:: python
+
+    from kartothek.io.iter import store_dataframes_as_dataset__iter
+
+    # "Generate" 5 dataframes
+    df_gen = (
+        pd.DataFrame({"date": pd.Timestamp(f"2020-01-0{i}"), "X": np.random.choice(10, 10)})
+        for i in range(1, 6)
+    )
+
+    dm = store_dataframes_as_dataset__iter(
+        df_gen,
+        store_factory,
+        "secondarily_indexed",
+        partition_on="date",
+        secondary_indices="X",
+    )
+    dm
+
+    dm = dm.load_all_indices(store_factory())
+    dm.indices["X"].eval_operator("==", 0)  # Show files where `X == 0`
+
+
+As can be seen from the example above, both ``partition_on`` and ``secondary_indices``
+can be specified together. Multiple ``secondary_indices`` can also be added as a list of
+strings.
+
+In general, secondary indices behave like explicit partitions in terms of when and how they
+can and cannot be created. However, when using ``partition_on`` the order of the columns
+provided is important, whereas it is ignored for ``secondary_indices``: this is because the
+underlying processes differ and creation of the :class:`~kartothek.core.index.PartitionIndex`
+is merely a side-effect of using ``partition_on``.
+
+
+Updating an existing dataset
+============================
+
+It's possible to update datasets by adding new physical partitions to them, ``kartothek``
+provides update functions that generally have the prefix `update_dataset` in their names.
+For example, :func:`~kartothek.io.eager.update_dataset_from_dataframes` is the update
+function for the ``eager`` backend.
+
+To see updating in action, let's set up some data and a storage location first and store
+the data there with ``kartothek``:
+
+.. ipython:: python
+
+    dm = store_dataframes_as_dataset(store_factory, "a_unique_dataset_identifier", df)
+    sorted(dm.partitions.keys())
+
+
+Now, we create ``another_df`` with the same schema as our intial dataframe
+``df`` and update it using the ``eager`` backend by calling :func:`~kartothek.io.eager.update_dataset_from_dataframes`:
+
+.. ipython:: python
+
+    from kartothek.io.eager import update_dataset_from_dataframes
+
+    another_df = pd.DataFrame(
+        {
+            "A": 5.0,
+            "B": pd.Timestamp("20110102"),
+            "C": pd.Series(2, index=list(range(4)), dtype="float32"),
+            "D": np.array([6] * 4, dtype="int32"),
+            "E": pd.Categorical(["prod", "dev", "prod", "dev"]),
+            "F": "bar",
+        }
+    )
+
+    dm = update_dataset_from_dataframes(
+        [another_df], store=store_factory, dataset_uuid="a_unique_dataset_identifier"
+    )
+    sorted(dm.partitions.keys())
+
+
+Looking at ``dm.partitions``, we can see that another partition has
```

Comment:
While the _add partitions_ piece is what's actually happening, I guess the user is much more interested in the data which is added to the dataset. I instruct people who write tests for datasets usually to *not* assert for given partitions but rather for the actual data to be there, e.g.

```
read_table(..., predicates=[[("A", "==", 5.0)]])
# print new row
...
```

Reply:
That's a few lines further. Should I delete the `dm.partitions` part?